src/bdfsclient/ActionHandler.cpp

```diff
@@ -78,8 +93,38 @@ namespace dfs
 		return 0;
 	}
 
+  void ActionHandler::Unmount(std::string nbdPath, bool matchAll=false)
+  {
+    std::ifstream fp("/proc/mounts");
+    std::string line,s1,s2;
+    while(fp >> s1 >> s2 && std::getline(fp,line))
+    {
+      if((s1.compare(0,nbdPath.length(),nbdPath) == 0 && matchAll) 
+        || s1 == nbdPath)
+      {
+        printf("Unmounting %s\n",s2.c_str());
+        std::string cmd = "umount " + s2;
+        system(cmd.c_str());
+      }
+    }
+    fp.close();
+  }
+
+  void ActionHandler::Cleanup()
+  {
+    printf("Unmounting volumes...\n");
+    ActionHandler::Unmount("/dev/nbd",true);
```

Comment:
Why don't we call UnbindVolume instead of creating a special case handling in Unmount?

Reply:
Each UnbindVolume call will open /proc/mounts and scan to find mount path for nbd.
Directly handling Unmount with special cause will only open and scan file once.
ubd_unbind is called after unmount.